dev_handler/dev_handler.h

```diff
@@ -0,0 +1,45 @@
+/**
+ * Handles lowcar device connects/disconnects and acts as the interface between the devices and shared memory
+ */
+
+#ifndef DEV_HANDLER_H
+#define DEV_HANDLER_H
+
+#include "../shm_wrapper/shm_wrapper.h"
+#include "message.h"
+#include "../runtime_util/runtime_util.h"
+#include "../logger/logger.h"
+
+#include <pthread.h>    // sender, receiver, and relayer threads
+#include <termios.h>    // POSIX terminal control definitions in serialport_init()
+#include <stdio.h>      // Print
+#include <stdlib.h>
+#include <stdint.h>     // ints with specified sizes (uint8_t, uint16_t, etc.)
+#include <signal.h>     // Used to handle SIGINT (Ctrl+C)
+#include <sys/time.h>   // For timestamps on PING
+#include <unistd.h>     // sleep(int seconds)
+
+// ************************************ CONFIG ************************************* //
+
+/* The maximum number of milliseconds to wait between each PING from a device
+ * Waiting for this long will exit all threads for that device (doing cleanup as necessary) */
+#define TIMEOUT 2500
+
+/* The number of milliseconds between each PING sent to the device */
+#define PING_FREQ 1000
+
+// ************************************ PUBLIC FUNCTIONS ************************************* //
+
+// Initialize logger, shm, and mutexes
+void init();
+
+// Disconnect devices from shared memory and destroy mutexes
+void stop();
+
+/**
+ * Detects when devices are connected
+ * On Arduino device connect, connect to shared memory and spawn three threads to communicate with the device
+ */
+void poll_connected_devices();
+
+#endif
```

Comment:
I moved `dev_handler.h` into `dev_handler.c`!

Though to address breaking up dev handler into separate files, the functions can be grouped into
1. Hotplugging
2. Receiver, sender, relayer
3. Sending/Receiving message
4. Serial port open/close

(2) includes `relay_clean_up()`, which works with (1) to coordinate which devices are new. (1) and (2) may stay...
(3) Is used only by functions in (2), but can be moved into `message.c` with some changes to function parameters. (At the moment, that file is only for building/parsing messages which made sense when we used `libusb`)
(4) Can be moved into a `serial_port` file, though it's going to be really short (around same length as `net_util.c`)

Reply:
Yeah, in that case I think that it might only make sense to move (3) into `message.c` then. But even that will help a bit, since those two functions together are probably like 100 lines ish?

Reply:
Actually I take it back, I think we should leave it :P `send_message` and `receive_message` both have things to do with the relays and the mutexes and whatnot that we have, and it would be bad to put them in `message.c` and break the abstraction barrier. All of the functions in `message.c` are working with bit and byte manipulation to pack and unpack the data, whereas all the functions in `dev_handler.c` are working with the all of the control logic with the devices, away from the abstraction of all the bits and bytes on the serial port. I like that organization, it's ok if the file is a bit long~~

Reply:
That's true! All `dev_handler.c` has to do is `send/receive_message()`, `make/destroy_message()`, `parse_message()`, and `read()/write()` without needing to know the messaging protocol. (Besides knowing to read the delimiter and then the cobs length)